tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -53,7 +57,7 @@ public void SubscribeToChanges(ISubscription subscription)
         {
             if (!_subscriptions.Contains(subscription))
             {
-                _subscriptions.Add(subscription);
+                _subscriptions = new List<ISubscription>(_subscriptions) { subscription };
```

Comment:
Totally missed the opportunity to use all the fancy C#12 bits here 😉 Ok... yes... you need to add the pragma too 😶 

```suggestion
                _subscriptions = [.. _subscriptions, subscription];
```

Reply:
I hesitated 😄  Added in https://github.com/DataDog/dd-trace-dotnet/pull/4996/commits/de6195058a6c95e9abc96b7febb56a266773db95